src/target/riscv/riscv-013.c

```diff
@@ -145,6 +145,62 @@ typedef struct {
 	struct target *target;
 } target_list_t;
 
+struct ac_cache {
+	uint32_t *commands;
+	size_t size;
+};
+
+static int ac_cache_elem_comparator(const void *p_lhs, const void *p_rhs)
+{
+	const uint32_t lhs = *(const uint32_t *)p_lhs;
+	const uint32_t rhs = *(const uint32_t *)p_rhs;
+	return (lhs > rhs) - (lhs < rhs);
```

Comment:
This is a creative way how to return the proper value. Still, I am afraid that it makes the code harder to grasp for readers.

For that reason, and if you don't mind, I'd prefer the (more boring but very clear) version with the `if`s:

```suggestion
	if (lhs < rhs)
		return -1;		
	if (lhs > rhs)
		return 1;
	return 0;
```

Reply:
Addressed